### DISCUSSION TITLE
Splitting Registries into different RC instances for Scaling independently

### DISCUSSION THREAD

**Discussion by parthlawate** [Sunbird-RC/community#292] [General]
Context : 
I've come across a use case where there are multiple registries created in one Sunbird RC instance. In order to scale each instance independently, they need to be to split into their own RC instances. 

Questions : 

1. Is this the right way to scale RC Or would you suggest an alternative mechanism where we need to control scaling per registry ? 
2. Whats the quickest way to do this migration ?

**Comment by parthlawate** [comment]
@tejash-jl could you help with this please ?

**Reply by tejash-jl** [reply]
@parthlawate any reason why you would need to split the existing registries into multiple instances?
If it was about handling scale, we can have RC instance setup on a K8 cluster, which can auto scale based on load. Also, have a production grade PSql cluster and ES cluster.

**Reply by parthlawate** [reply]
here the use case is about being able to scale some registries & not having to scale everything..  from a net cost perspective make it more cost efficient .. and also allow hosting each registry across different cloud providers. 
not sure if thats a valid way to look at it though..

**Reply by pramodkvarma** [reply]
Not sure @parthlawate . Co-hosting multiple using shared infra maybe more cost effective.

**Reply by parthlawate** [reply]
@coolbung can you share more insights on the 'WHY' here  ?

**Reply by coolbung** [reply]
There are 3 key groups of registries. Each registry has varying number of schemas.

Of which, the candidate registry will have the most traffic (say over 100x the traffic of the other registries)

We have one commercial postgres license. So the plan is to use the commercial license for the user registry+keycloak and the other registries can run on the community license.

For the remaining 2 registries, splitting is for better schema and traffic management. One of these 2 has data that will be connected to an e-commerce system. So the separation will help us have better fault tolerance, or even move the e-commerce linked registry to a commercial postgres license.

**Comment by pramodkvarma** [comment]
OK, seems doable. Is the commercial license tied to VM cores? If so, what you are saying makes sense. Btw, UPI runs on community license (of course, heavily front ended by Radis clusters)!!
